tools/roxie/extract-roxie-timings.py

```diff
@@ -23,35 +23,95 @@
 import sys
 import re
 import argparse
+import datetime
 
 def calculateDerivedStats(curRow):
 
     numBranchHits = float(curRow.get("NumNodeCacheHits", 0.0))
     numBranchAdds = float(curRow.get("NumNodeCacheAdds", 0.0))
     numBranchFetches = float(curRow.get("NumNodeDiskFetches", 0.0))
     timeBranchFetches = float(curRow.get("TimeNodeFetch", 0.0))
+    timeBranchRead = float(curRow.get("TimeNodeRead", 0.0))
+    timeBranchLoad = float(curRow.get("TimeNodeLoad", 0.0))
+    timeBranchDecompress = timeBranchLoad - timeBranchRead
+    avgTimeBranchDecompress = 0 if numBranchAdds == 0 else timeBranchDecompress / numBranchAdds
+
     if numBranchHits + numBranchAdds:
         curRow["%BranchMiss"] = 100*numBranchAdds/(numBranchAdds+numBranchHits)
 
     numLeafHits = float(curRow.get("NumLeafCacheHits", 0.0))
     numLeafAdds = float(curRow.get("NumLeafCacheAdds", 0.0))
     numLeafFetches = float(curRow.get("NumLeafDiskFetches", 0.0))
     timeLeafFetches = float(curRow.get("TimeLeafFetch", 0.0))
+    timeLeafRead = float(curRow.get("TimeLeafRead", 0.0))
+    timeLeafLoad = float(curRow.get("TimeLeafLoad", 0.0))
+    timeLeafDecompress = timeLeafLoad - timeLeafRead
+    avgTimeLeafDecompress = 0 if numLeafAdds == 0 else timeLeafDecompress / numLeafAdds
+
+    timeLocalExecute = float(curRow.get("TimeLocalExecute", 0.0))
+    timeAgentWait = float(curRow.get("TimeAgentWait", 0.0))
+    timeAgentProcess = float(curRow.get("TimeAgentProcess", 0.0))
+    timeSoapcall = float(curRow.get("TimeSoapcall", 0.0))
+
+    agentRequestEstimate = numLeafHits + numLeafAdds                                  # This could be completely wrong, but better than nothing
+    numAgentRequests = float(curRow.get("NumAgentRequests", agentRequestEstimate))    # 9.8.x only
+
+    timeLocalCpu = timeLocalExecute - timeAgentWait - timeSoapcall
```

Comment:
timeAgentWait and timeSoapcall are [parallel] aggregate sums and ok this is an approx but this could possibly go -ve